### ISSUE TITLE
Communication between components

### ISSUE THREAD

**Issue by jscriptcoder** [elm-zurich/meetup#7]
Hi there,

this is something I ran into when playing around with Elm. Currently I'm working on a personal project, quite componentized (if that word even exists), where I need to pass information from child to parent (sometimes three levels up), child to child or even component to component that aren't in the same branch. I figured out a few possible solutions, but it would be great to have a talk about it since I don't even know if that's the right way or there is a better one, sort of standard way.

Thanks in advance

**Comment by stoeffel** [comment]
hey @jscriptcoder!
maybe reading https://guide.elm-lang.org/reuse/ wil help you.
cheers

**Comment by ccapndave** [comment]
For my latest project I'm using a flat hierarchy made with extensible records which means that most of the time you simply don't need to communicate from child to parent (Noah at NoRedInk came up with this, btw, not me :) )  I don't think its the standard way, but I've found that it works really well; I'll happily show you tonight.

**Comment by stoeffel** [comment]
Yes, we use that sometimes. But it's also okay to just pass parts of the model to the child view.

One pattern I recently like is (without extensible records):

``` elm

select : Model -> SubModel
select model =
  { disabled = List.isEmpty model.users
  , userCount = List.length model.users
  }

subView : SubModel -> Html msg
subView model =
  -- display stuff

-- in parent
subView (select model)
```

**Comment by jscriptcoder** [comment]
Great... thanks guys. Looking forward to tonight's meetup
